### ISSUE TITLE
Some unit tests are flaky due to reliance on LockSupport.getBlocker()

### ISSUE THREAD

**Issue by jsampson** [jsampson/guarded-executor#3]
I've seen some of the tests in `SupplementalGuardedExecutorTest` failing occasionally (seems to be ~1% chance). The failures indicate that a thread had unblocked and executed its own task even though it had already parked as indicated by `LockSupport.getBlocker()` (called by `TestingThread.waitForParked()`).

I'm comfortable relying on `getBlocker()` to indicate that a thread had definitely parked. It doesn't have specific documented memory ordering semantics, so we shouldn't treat it as a happens-before relation, but it can't possibly be set unless the thread in question definitely called `park()`.

However, `park()` itself can return spuriously, which means that we can't assume that the thread will _remain_ parked for any length of time after calling `getBlocker()`. Therefore we need some other way to ensure that the thread in question doesn't execute its own task even if it does unblock and acquire the lock unexpectedly.
